src/coreclr/vm/jitinterface.cpp

```diff
@@ -1570,15 +1573,31 @@ void CEEInfo::getFieldInfo (CORINFO_RESOLVED_TOKEN * pResolvedToken,
 
 #ifdef HOST_WINDOWS
 #ifndef TARGET_ARM
-                bool canOptimizeHelper = (pResult->helper == CORINFO_HELP_GETSHARED_NONGCTHREADSTATIC_BASE_NOCTOR) ||
-                    (pResult->helper == CORINFO_HELP_GETSHARED_NONGCTHREADSTATIC_BASE);
-                // For windows, we convert the TLS access to the optimized helper where we will store
-                // the static blocks in TLS directly and access them via inline code.
-                if (canOptimizeHelper && ((pField->GetFieldType() >= ELEMENT_TYPE_BOOLEAN) && (pField->GetFieldType() < ELEMENT_TYPE_STRING)))
+                bool canOptimizeHelper =
```

Comment:
@jkotas - Is this a good set of types to cover the GC types? Is there a better way to check for non-generics GC types that we can optimize?

Reply:
Why do you need to do the filtering? Can you enable this for all types?

Reply:
I did that initially, but I was hitting AV in most of the libraries tests and hence added the filter. Shouldn't I be restricting it for `ELEMENT_TYPE_GENERICINST` for instance?

Reply:
You should never get `ELEMENT_TYPE_GENERICINST` from `GetFieldType()` here.

Reply:
Ok. Let me add back other types and try to narrow down which type was causing the problem.

Reply:
Fixed. Turned out this was gc-info reporting bug. Thanks @jkotas .